### ISSUE TITLE
Lint fails and states that an old version of Node.js is in use (with Node installed via n version management)

### ISSUE THREAD

**Issue by DMeechan** [vuejs/vue-cli#1723]
### Version
3.0.0-rc.3

### Reproduction link
[https://github.com/DMeechan/meechan-co](https://github.com/DMeechan/meechan-co)

### Steps to reproduce
Summary: 

1. Install latest version of Node (v10.5.0) using n
2. Create a Vue project with linting enabled on commit
3. When trying to create a commit, the linter fails and erroneously reports that I'm using Node v7.10.0

Detailed steps:

1. `yarn global add n`
2. `n latest` (installs Node v10.5.0 at time of writing)
3. `yarn global add @vue/cli`
4. `vue create sample-project`
5. Select `Manually select features`
6. Select `Linter / Formatter`
7. Select `ESLint   Prettier`
8. Select both `Lint on save` and `Lint and fix on commit`
9. Select `In package.json`
10. `cd sample-project`
11. `yarn`
12. `git add .`
13. `git commit -m "init"`

Error log appears (see below).

### What is expected?
The linting and commit to succeed.

### What is actually happening?
When attempting to create a commit, the linter fails and displays this error message:

```text
 > running pre-commit hook: lint-staged
 ❯ Running tasks for *.js
   ✖ vue-cli-service lint
     → Please upgrade your Node version.
     git add
 ❯ Running tasks for *.vue
   ✖ vue-cli-service lint
     → Please upgrade your Node version.
     git add
✖ vue-cli-service lint found some errors. Please fix them and try committing again.

 ERROR  You are using Node v7.10.0, but vue-cli-service requires Node >=8.
Please upgrade your Node version.
✖ vue-cli-service lint found some errors. Please fix them and try committing again.

 ERROR  You are using Node v7.10.0, but vue-cli-service requires Node >=8.
Please upgrade your Node version.

pre-commit hook failed (add --no-verify to bypass)
```

---
Running `node --version` shows that I'm running Node v10.5.0.

**Comment by LinusBorg** [comment]
While we still strugle for a setup to reproduce this in, can you quickly run:

1. `npm run lint`
2. `node_modules/.bin/eslint /src` 

So test if this error is at al related to vue-cli? 

* If it is vue-cli, only the second command should run fine, while the first should throw an error. 
* If the first runs fine as well, the problem is more likely with the git hooks/package running those hooks (lint-staged).

**Comment by LinusBorg** [comment]
can you do the following?

1. Create a small file called test.js anywhere. It should look like this:
```javascript
 #!/usr/bin/env node

console.log(process.version)
```
2. make the file an executable:
```
chmode +x test.js
```
3. Run the executable

```
./test.js
```

My supicion is that this is a misconfiguration in your environment rather than a problem with vue-cli, and I would like to disprove that before risking to wreck my own env by installing `n` and `nvm` in parallel or anything.

**Comment by sodatea** [comment]
Closing due to inactivity. Please open a new issue with a reference to this one if you can follow up with more information.

**Comment by wushq5** [comment]
i reproduce it by upgrade `@vue/cli-service` from `v3.0.0-beta.6` to  `v3.0.1`，when i run vue-cli-service lint on commit, the console displays errors as below
```
 > running pre-commit hook: lint-staged
 ❯ Running tasks for src/**/*.js
   ✖ vue-cli-service lint
     git add
 ↓ Running tasks for src/**/*.vue [skipped]
   → No staged files match src/**/*.vue
✖ "vue-cli-service lint" found some errors. Please fix them and try committing again.

 ERROR  You are using Node v6.11.0, but vue-cli-service requires Node >=8.
Please upgrade your Node version.
```

note that, my node version is `v8.11.4` installed via nvm.
